cmd/registry/root.go

```diff
@@ -0,0 +1,261 @@
+		providerId, nexists := providerJson["name"].(string) // displayName??
+		logKeyNotExists(nexists, "name", providerJson)
+		log.Infof("Adding provider required for DAG: %s", providerId)
+		addProviderByName(providerId)
+	}
+}
+
+func getDagRoute(dagId string, dagVersion string) string {
+	filledDagRoute := fmt.Sprintf(dagRoute, dagId, dagVersion)
+	getDagUrl := url.URL{
+		Scheme: "https",
+		Host:   registryHost,
+		Path:   fmt.Sprintf("%s/%s", registryApi, filledDagRoute),
+	}
+	return getDagUrl.String()
+}
+
+func downloadDagToFile(sourceUrl string, path string) {
+	file, err := create(path)
+	client := http.Client{
+		CheckRedirect: func(r *http.Request, via []*http.Request) error {
+			r.URL.Opaque = r.URL.Path
+			return nil
+		},
+	}
+	resp, err := client.Get(sourceUrl)
+	logFatal(err)
+	defer resp.Body.Close()
+	_, err = io.Copy(file, resp.Body)
+	defer file.Close()
+	log.Infof("Downloaded file %s from %s", path, sourceUrl)
+}
+
+func newRegistryCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "registry",
+		//Aliases: []string{"r"},
+		Short: "Utilize the Astronomer Registry",
+		Long:  "Utilize the Astronomer Registry to add a DAG or a Provider.",
+	}
+	cmd.AddCommand(
+		newRegistryAddCmd(),
+		newRegistryInitCmd(),
+	)
+	return cmd
+}
+
+func newRegistryInitCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "init",
+		//Aliases: []string{"r"},
+		Short: "Utilize the Astronomer Registry",
+		Long:  "Utilize the Astronomer Registry to initialize a project from a template.",
+	}
+	//cmd.AddCommand(
+	//	newRegistryAddDagCmd(),
+	//	newRegistryAddProviderCmd(),
+	//)
+	return cmd
+}
+
+func newRegistryAddCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "add",
+		//Aliases: []string{"r"},
+		Short: "Utilize the Astronomer Registry",
+		Long:  "Utilize the Astronomer Registry to add a DAG or a Provider.",
+	}
+	cmd.AddCommand(
+		newRegistryAddDagCmd(),
+		newRegistryAddProviderCmd(),
+	)
+	return cmd
+}
+
+func newRegistryAddDagCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "dag [DAG ID]",
+		//Aliases: []string{"d"},
+		Short: "Utilize the Astronomer Registry to add a DAG",
+		Long:  "Utilize the Astronomer Registry to download a DAG to your local project.",
+		Run: func(cmd *cobra.Command, args []string) {
+			var dagId string
+
+			// if a dagId was provided in the args we use it
+			if len(args) > 0 {
+				dagId = args[0]
+			} else {
+				// no dagId was provided so ask the user for it
+				dagId = input.Text("enter DAG ID to download: ")
+			}
+			downloadDag(dagId, dagVersion)
+		},
+	}
+	cmd.Flags().StringVar(&dagVersion, "version", "", "Optional DAG Version to Download.")
+	return cmd
+}
+
+func newRegistryAddProviderCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use: "provider [PROVIDER]",
+		//Aliases: []string{"p"},
+		Short: "Utilize the Astronomer Registry to add a Provider",
+		Long:  "Utilize the Astronomer Registry to download a Provider like Snowflake to your project's dependencies.",
```

Comment:
Perhaps reword the "like Snowflake" into (like Snowflake or Great Expectations)... super open to other suggestions!